### PR TITLE
 fab script for python3 projects

### PR DESCRIPTION
**Pull request by gavinwahl**
Creates a python2 virtualenv just for fabric.

### CODE REVIEW

**Comment by gavinwahl** [comment]
You can make this easy to use by doing `alias fab='PATH=$PATH:. "fab"'`

File: fab [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+HELPERS_COMMIT=f854d6a00
+HELPERS_COMMIT_FILE=$ENV/HELPERS_COMMIT
+
+if [ ! -d $ENV ]
+then
+  virtualenv -p `which python2` $ENV
+  touch $HELPERS_COMMIT_FILE
+fi
+source $ENV/bin/activate
+if [ "$HELPERS_COMMIT" != "`cat $HELPERS_COMMIT_FILE`" ]
```
Comment (acatton) [inline fab]: Can we use `$()` instead of back-ticks. This is not cargo cult, I find `$()` to be better style, back-ticks feels 70s.

File: fab [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#! /bin/bash
+
+ENV=.deployment_virtualenv
+# version of fusionbox-fabric-helpers to run
+HELPERS_COMMIT=f854d6a00
+HELPERS_COMMIT_FILE=$ENV/HELPERS_COMMIT
+
+if [ ! -d $ENV ]
+then
+  virtualenv -p `which python2` $ENV
```
Comment (acatton) [inline fab]: Isn't `virtualenv` looking up the path. Isn't `virtualenv -p python2 $ENV` doing the same thing? (This is bike-shedding)

File: fab [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  touch $HELPERS_COMMIT_FILE
+fi
+source $ENV/bin/activate
+if [ "$HELPERS_COMMIT" != "`cat $HELPERS_COMMIT_FILE`" ]
+then
+  pip install fabric -e "git+https://github.com/fusionbox/fusionbox-fabric-helpers@$HELPERS_COMMIT#egg=fusionbox-fabric-helpers" >/dev/null
+  echo $HELPERS_COMMIT > $HELPERS_COMMIT_FILE
+fi
+exec fab "$@"
```
Comment (acatton) [inline fab]: Yes! :+1: `exec`.

File: fab (hunk 2 above)
Comment (gavinwahl) [inline fab]: No

File: fab (hunk 1 above)
Comment (gavinwahl) [inline fab]: Why?

Comment (acatton) [inline fab]: Visually backticks can be confused with single quotes, also it makes it hard to parse for human. Expensions in strings is done by `$...` with the exception of backticks. When seeing a `$` sign people go "oh... this is gonna get expended", but with backticks (I mean it was the case for me) people go "what did he mean here? wait.. there are backticks... I see"
